JustSaying/AwsTools/MessageHandling/SqsNotificationListener.cs

```diff
@@ -143,53 +135,51 @@ private static string AggregateExceptionDetails(AggregateException ex)
             return innerExDetails.ToString();
         }
 
-        public void StopListening()
-        {
-            _cts.Cancel();
-            _log.LogInformation($"Stopping Listening - Queue: {_queue.QueueName}, Region: {_queue.Region.SystemName}");
-        }
-
         internal async Task ListenLoop(CancellationToken ct)
         {
             var queueName = _queue.QueueName;
             var region = _queue.Region.SystemName;
             ReceiveMessageResponse sqsMessageResponse = null;
 
-            try
+            while (!ct.IsCancellationRequested)
             {
-                sqsMessageResponse = await GetMessagesFromSqsQueue(ct, queueName, region).ConfigureAwait(false);
-                var messageCount = sqsMessageResponse.Messages.Count;
+                try
+                {
+                    sqsMessageResponse = await GetMessagesFromSqsQueue(ct, queueName, region).ConfigureAwait(false);
+                    var messageCount = sqsMessageResponse.Messages.Count;
 
-                _log.LogTrace($"Polled for messages - Queue: {queueName}, Region: {region}, MessageCount: {messageCount}");
-            }
-            catch (InvalidOperationException ex)
-            {
-                _log.LogTrace($"Could not determine number of messages to read from [{queueName}], region: [{region}]. Ex: {ex}");
-            }
-            catch (OperationCanceledException ex)
-            {
-                _log.LogTrace($"Suspected no message in queue [{queueName}], region: [{region}]. Ex: {ex}");
-            }
-            catch (Exception ex)
-            {
-                var msg = $"Issue receiving messages for queue {queueName}, region {region}";
-                _log.LogError(0, ex, msg);
-            }
+                    _log.LogTrace(
+                        $"Polled for messages - Queue: {queueName}, Region: {region}, MessageCount: {messageCount}");
+                }
+                catch (InvalidOperationException ex)
+                {
+                    _log.LogTrace(
+                        $"Could not determine number of messages to read from [{queueName}], region: [{region}]. Ex: {ex}");
```

Comment:
This can only assume the reason for `InvalidOperationException` because it knows what happens inside `GetNumberOfMessagesToReadFromSqs()`. It shouldn't be assuming what happens inside other methods. Better to just let this get caught by the catch-all block.